ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/TopAppBar.kt

```diff
@@ -285,15 +349,64 @@ private fun TopBarOneVisibleActionMenuExample(
     )
 }
 
+@Composable
+fun TopAppBarBackNavigation(
+    backContentDescriptionText: String? = null,
+    backIconVector: ImageVector = Icons.AutoMirrored.Filled.ArrowBack,
+    backText: String? = null,
+    onBack: () -> Unit
+) {
+    Box(
+        modifier =
+            Modifier
+                .wrapContentSize()
+                .clip(RoundedCornerShape(ZcashTheme.dimens.regularRippleEffectCorner))
+                .clickable { onBack() }
+    ) {
+        Row(
+            modifier = Modifier.padding(all = ZcashTheme.dimens.spacingDefault),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Icon(
+                imageVector = backIconVector,
+                contentDescription = backContentDescriptionText,
+            )
+
+            backText?.let {
+                Spacer(modifier = Modifier.size(size = ZcashTheme.dimens.spacingSmall))
+
+                Text(text = backText)
+            }
+        }
+    }
+}
+
+@Composable
+fun TopAppBarHideBalancesNavigation(
+    iconVector: ImageVector,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    contentDescription: String? = null,
+) {
+    IconButton(
+        onClick = onClick,
+        modifier = modifier
+    ) {
+        Icon(
+            imageVector = iconVector,
+            contentDescription = contentDescription,
+            modifier = Modifier.size(24.dp)
```

Comment:
icon is of size 24.dp by default already https://developer.android.com/develop/ui/compose/graphics/images/material

Reply:
The injected vector resources are smaller here and do not stretch automatically if this is what you mean?